lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/__tests__/NxCollapsibleMultiSelect.test.tsx

```diff
@@ -4,355 +4,559 @@
+          const trigger = renderAndGetTrigger({
+            disabled: true,
+            disabledTooltip: 'No options'
+          });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          const tooltip = screen.getByRole('tooltip');
+
+          expect(tooltip).toHaveTextContent('No options');
+        });
+
+        it('has specified classname when tooltipModifierClass prop is provided', async function() {
+          const trigger = renderAndGetTrigger({
+            disabled: true,
+            disabledTooltip: 'No options',
+            tooltipModifierClass: 'tooltipClass'
+          });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          const tooltip = screen.getByRole('tooltip');
+
+          expect(tooltip).toHaveTextContent('No options');
+          expect(tooltip.querySelector('.tooltipClass')).toBeInTheDocument();
+        });
+
+        it('does not show when disabledTooltip prop is not provided', async function() {
+          const trigger = renderAndGetTrigger({ disabled: true });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+        });
+      });
     });
+  });
 
-    it('uses option name as checkbox label', function () {
-      const renderOption = getShallow().renderProp('renderOption');
+  describe('filter', function() {
+    const filterView = (props?: Partial<Props>) => quickRender({
+      ...props,
+      filterThreshold: 2,
+      onFilterChange: jest.fn()
+    });
+
+    it('renders an input with type="text" iff when the number of options ' +
+    'is greater than filterThreshold and onFilterChange is supplied', function() {
+      const noFilterView = quickRender({ onFilterChange: jest.fn() });
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('children', 'Foo');
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('children', 'Bar');
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('children', 'Null');
+      expect(quickRender().queryByRole('textbox')).not.toBeInTheDocument();
+      expect(noFilterView.queryByRole('textbox')).not.toBeInTheDocument();
+      expect(filterView().getByRole('textbox')).toHaveAttribute('type', 'text');
     });
 
-    it('generates checkboxId using name prop and option name', function () {
-      const shallowRender = getShallow();
-      const renderOption = shallowRender.renderProp('renderOption');
+    it('sets the filter input placeholder as specified by filterPlaceholder, ' +
```

Comment:
Should check the a11y name too

Reply:
As discussion, the RTL currently does not consider placeholder as an accessible name https://github.com/testing-library/dom-testing-library/issues/926 while it is defined in [HTML Accessibility API Mappings 1.0](https://www.w3.org/TR/html-aam-1.0/#input-type-text-input-type-password-input-type-number-input-type-search-input-type-tel-input-type-email-input-type-url-and-textarea-element-accessible-name-computation)(working draft), before RTL has any further updates, will just have the tests for checking it has placeholder attrs.